pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/RRDD.scala

```diff
@@ -312,6 +314,27 @@ private class StringRRDD[T: ClassTag](
   lazy val asJavaRDD : JavaRDD[String] = JavaRDD.fromRDD(this)
 }
 
+private class BufferedStreamThread(in: InputStream, name: String) extends Thread(name) {
+  val ERR_BUFFER_SIZE = 100
+  val lines = new Array[String](ERR_BUFFER_SIZE)
+  var lineIdx = 0
+  override def run() {
+    for (line <- Source.fromInputStream(in).getLines) {
```

Comment:
It seems to me this can be simplified to something like:
```
val output = Source.fromInputStream(in).getLines()
output.foreach(println)
lines = output.takeRight(ERR_BUFFER_SIZE)
```